src/commands/install/install.py

```diff
@@ -0,0 +1,48 @@
+from enum import Enum
+from os import path
+from typing import Optional
+from git import InvalidGitRepositoryError, NoSuchPathError, Repo
+from git.objects import Submodule
+from typing_extensions import TypedDict
+from .utils import map_url_to_package_name, verify_repo_url
+
+
+class InstallationErrorEnum(Enum):
+    INCORRECT_URL = "INCORRECT_URL"
+    INVALID_LOCAL_REPOSITORY = "INVALID_LOCAL_REPOSITORY"
+    INVALID_PACKAGE_NAME = "INVALID_PACKAGE_NAME"
+
+
+InstallationResultType = TypedDict(
+    "InstallationResult", {"error": Optional[InstallationErrorEnum]}
+)
+
+
+def install(
+    url: str,
```

Comment:
We probably should allow just repo name as the parameter, and perhaps an optional full URL if we want to support installing from other sources than github

Reply:
I'll add that option in the future. The repo name is not enough, it must be account-name/repo-name. Pasting a URL is easier than manually extracting package name that's why I focus on this approach.

Reply:
https://github.com/software-mansion/protostar/issues/7